lib/src/system/system_manager.dart

```diff
@@ -1,28 +1,50 @@
 part of oxygen;
 
+/// Manages all registered systems.
 class SystemManager {
+  /// The world in which this manager lives.
   final World world;
 
-  List<System> _systems = [];
+  /// All the registered systems.
+  final List<System> _systems = [];
+
+  final Map<Type, System> _systemsByType = {};
 
   SystemManager(this.world);
 
+  /// Initialize all the systems that are registered.
   void init() {
     for (final system in _systems) {
       system.init();
     }
   }
 
+  /// Register a system.
   void registerSystem<T extends System>(T system) {
     assert(system.world == null, '$T is already registered');
-    system.world = this.world;
+    system.world = world;
     _systems.add(system);
+    _systemsByType[T] = system;
+
+    _systems.sort((a, b) => a.priority - b.priority);
+  }
+
+  /// Unregister a system.
+  void unregisterSystem(Type systemType) {
```

Comment:
deregisterSystem?
https://english.stackexchange.com/questions/25931/unregister-vs-deregister

Reply:
Agreed, I also updated the docs to reflect this change.